common/coin_support/solana.h

```diff
@@ -0,0 +1,114 @@
+/**
+ * @file    solana.h
+ * @author  Cypherock X1 Team
+ * @brief   Header for solana coin.
+ *          Stores declarations for solana coin functions.
+ * @copyright Copyright (c) 2022 HODL TECH PTE LTD
+ * <br/> You may obtain a copy of license at <a href="https://mitcc.org/" target=_blank>https://mitcc.org/</a>
+ *
+ */
+#ifndef SOLANA_HEADER
+#define SOLANA_HEADER
+
+#include <stdint.h>
+#include "coin_utils.h"
+
+// Derivation path reference : https://docs.solana.com/wallet-guide/paper-wallet#hierarchical-derivation
+#define SOLANA_PURPOSE_INDEX 0x8000002C
+#define SOLANA_COIN_INDEX    0x800001F5
+#define SOLANA_ACCOUNT_INDEX 0x80000000
+#define SOLANA_CHAIN_INDEX   0x80000000
+#define SOLANA_ADDRESS_INDEX 0x80000001
+
+#define SOLANA_ACCOUNT_ADDRESS_LENGTH 32
+#define SOLANA_BLOCKHASH_LENGTH       32
+
+// Reference : https://docs.rs/solana-program/1.14.3/solana_program/system_instruction/enum.SystemInstruction.html#variant.Transfer
+typedef struct solana_transfer_data {
+  uint8_t *funding_account;
+  uint8_t *recipient_account;
```

Comment:
Instead of adding a pointer to the address as a member of transfer, add a getter API for getting the recipient based on the first element of opaque data.

Reply:
not needed. [same comment](https://github.com/Cypherock/x1_wallet_firmware/pull/39#discussion_r986780563) applicable